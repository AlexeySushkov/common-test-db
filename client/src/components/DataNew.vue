<template>
  <v-main>
    <v-card
      class="mx-auto"
      outlined
      elevation="3"
      max-width="800"
    >
      <v-progress-linear
        :active="showProgress"
        :indeterminate="showProgress"
        :color = "configColor"
        absolute
        bottom
      ></v-progress-linear>
      <v-toolbar flat dense :color = "configColor" dark>
        <v-toolbar-title>New Data</v-toolbar-title>
      </v-toolbar>
      <v-container>
        <v-col>
          <v-select
            :items="['10', '11', '12']"
            label="Counter 1"
            v-model="newData.data.Counter1"
          ></v-select> 
          <v-select
            :items="['20', '21', '22']"
            label="Counter 2"
            v-model="newData.data.Counter2"
          ></v-select> 
        </v-col>
      </v-container>
      <v-card-actions>
        <v-btn
          class="newDataCall"
          rounded
          dark
          :color = "configColor"
          @click="newDataCall">
          Create
        </v-btn>     
        <v-btn
          class="Cancel"
          rounded
          dark
          :color = "configColor"
          @click="navigateTo({name: 'Home'})">
          Cancel
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-main>
</template>

<script>
import config from '../config'
import serverApi from '@/services/ApiCalls'

export default {
  data () {
    return {
      configColor: config.titleColor,
      showProgress: false,
      error: 'Status: ',

      newData: {
        data: {
          Counter1: '10',
          Counter2: '20'
        }
      }     
    }
  },

  methods: {
    async navigateTo (route) {
      this.$router.push(route)
    },
    async newDataCall () {
      console.log('newDataCall')

      this.showProgress = true

      try {
        const response = await serverApi.newData(this.newData)
        this.error = 'Status: ' + response.data.status
      } catch (error) {
        if ((error.response !== undefined) && (error.response.data.error !== undefined)) {
          this.error = error.response.status + ': ' + error.response.data.error
        } else {
          this.error = 'No connection to the server'
        }
      }
      this.showProgress = false
      alert(this.error)
    }
  },
  async mounted () {
    console.log('mounted newData isUserLoggedIn: ', this.$store.state.isUserLoggedIn)
    if (!this.$store.state.isUserLoggedIn) {
      this.navigateTo({ name: 'Home' }) 
    } 
  }

}
</script>

<style scoped>
.v-btn {
  text-transform: none;
}
</style>
