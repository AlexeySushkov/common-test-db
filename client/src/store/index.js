import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    token: null,
    email: null,
    sessionId: null,
    isUserLoggedIn: false,
    drawer: true,
    titleColor: 'blue darken-3'
  },
  mutations: {
    setToken (state, token) { 
      state.token = token
      localStorage.token = token
      if (token) {
        state.isUserLoggedIn = true
      } else {
        state.isUserLoggedIn = false
      }
    }, 
    setSessionId (state, id) { 
      state.sessionId = id
      sessionStorage.sessionId = id
    },     
    setEmail (state, email) { 
      state.email = email
    },
    setDrawer (state, _drawer) { 
      state.drawer = _drawer
    },
    setTitleColor (state, titleColor) { 
      state.titleColor = titleColor
      localStorage.titleColor = titleColor
    }
  },
  actions: {
    setEmail ({ commit }, email) { 
      commit('setEmail', email)
    },
    setToken ({ commit }, token) { 
      commit('setToken', token)
    },
    setSessionId ({ commit }, id) { 
      commit('setSessionId', id)
    },    
    setDrawer ({ commit }, drawer) { 
      commit('setDrawer', drawer)
    },
    setTitleColor ({ commit }, titleColor) { 
      commit('setTitleColor', titleColor)
    }
  },
  modules: { 
  } 
})
