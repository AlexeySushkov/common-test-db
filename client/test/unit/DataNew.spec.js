import Vue from 'vue';
import Vuex from 'vuex';
import Vuetify from 'vuetify'

import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import DataNew from '@/components/DataNew';
import store from '@/store/index'
const VueWithVuex = createLocalVue()
VueWithVuex.use(Vuex)
Vue.use(Vuetify)

describe('DataNew', () => {
  const localVue = createLocalVue()
  localVue.use(Vuetify)
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('DataNew wrapper', () => {
      const wrapper = mount(DataNew, {
        localVue,
        vuetify,
        store
      })
      expect(wrapper.html()).toMatchSnapshot()
  })
//   test('DataNew mount', () => {
//     jest.useFakeTimers()
//     const mountedCall = jest.spyOn(DataNew, 'mounted')
//     const wrapper = mount(DataNew, {
//       localVue: VueWithVuex,
//       store
//     })
//     console.log('this.error: ', wrapper.vm.error)
//     expect(mountedCall).toHaveBeenCalled()
//   })
//   test('DataNew store', () => {
//     const wrapper = mount(DataNew, {
//       localVue: VueWithVuex,
//       store
//     })
//     // this.$store.state.isUserLoggedIn
//     console.log('isUserLoggedIn before: ', store.state.isUserLoggedIn)
//     // this.$store.dispatch('setToken', response.data.token)
//     store.dispatch('setToken', 'Token from UnitTest')
//     console.log('isUserLoggedIn after: ', store.state.isUserLoggedIn)
// })
})

