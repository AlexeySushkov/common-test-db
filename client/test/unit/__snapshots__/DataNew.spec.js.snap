// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataNew DataNew wrapper 1`] = `
<main class="v-main v-content" style="padding: 0px 0px 0px 0px;">
  <div class="v-main__wrap v-content__wrap">
    <div class="mx-auto v-card v-sheet v-sheet--outlined theme--light elevation-3" style="max-width: 800px;">
      <div role="progressbar" aria-valuemin="0" aria-valuemax="100" aria-valuenow="0" class="v-progress-linear v-progress-linear--absolute theme--light" style="bottom: 0px; height: 0px;">
        <div class="v-progress-linear__background blue darken-3" style="opacity: 0.3; left: 0%; width: 100%;"></div>
        <div class="v-progress-linear__buffer" style="height: 0px;"></div>
        <transition-stub name="slide-x-transition">
          <div class="v-progress-linear__determinate blue darken-3" style="width: 0%;"></div>
        </transition-stub>
      </div>
      <header class="v-sheet theme--dark v-toolbar v-toolbar--dense v-toolbar--flat blue darken-3" style="height: 48px;">
        <div class="v-toolbar__content" style="height: 48px;">
          <div class="v-toolbar__title">New Data</div>
        </div>
      </header>
      <div class="container">
        <div class="col">
          <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
            <div class="v-input__control">
              <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-9" class="v-input__slot">
                <div class="v-select__slot"><label for="input-9" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Counter 1</label>
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">10</div><input id="input-9" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="10">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
          <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
            <div class="v-input__control">
              <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-14" class="v-input__slot">
                <div class="v-select__slot"><label for="input-14" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Counter 2</label>
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">20</div><input id="input-14" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                  </div><input type="hidden" value="20">
                </div>
                <div class="v-menu">
                  <!---->
                </div>
              </div>
              <div class="v-text-field__details">
                <div class="v-messages theme--light">
                  <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-card__actions"><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg v-btn--rounded theme--dark v-size--default blue darken-3"><span class="v-btn__content">
        Create
      </span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg v-btn--rounded theme--dark v-size--default blue darken-3"><span class="v-btn__content">
        Cancel
      </span></button></div>
    </div>
  </div>
</main>
`;
